storage/bootstrap/result/result.go

```diff
@@ -65,8 +65,22 @@ func (r *bootstrapResult) AddResult(other BootstrapResult) {
 	if other == nil {
 		return
 	}
-	r.results.AddResults(other.ShardResults())
-	r.unfulfilled.AddRanges(other.Unfulfilled())
+	size := 0
+	sizeOther := 0
+	for _, sr := range r.results {
+		size += len(sr.AllSeries())
+	}
+	for _, sr := range other.ShardResults() {
+		sizeOther += len(sr.AllSeries())
+	}
+	if size >= sizeOther {
+		r.results.AddResults(other.ShardResults())
+		r.unfulfilled.AddRanges(other.Unfulfilled())
+	} else {
+		other.AddResult(r)
```

Comment:
I see, perhaps document this behavior more explicitly?

Alternatively we could follow c++'s convention and make this method a non-member method that takes two results as the arguments and returns a result
```
MergeResults(result1, result2 BootstrapResult) BootstrapResult
```
Because this is not a member method, users are less inclined to assume this affects the object on which the method is called.